repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -161,40 +184,74 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
 
     // If we are not in a mutable mode, immediately send error to the wrapped response.
     if (headerMode != ResponseMode.MUTABLE && bodyMode != ResponseMode.MUTABLE) {
-      originalResponse.sendError(i, s)
+      sendErrorFunc
     }
   }
 
-  def isError: Boolean = sentError
+  def isError: Boolean = {
+    logger.trace("isError called")
+    sentError
+  }
 
-  def getReason: String = reason.orNull
+  def getReason: String = {
+    logger.trace("getReason called")
+    reason.orNull
+  }
 
-  override def isCommitted: Boolean = super.isCommitted || committed
+  override def isCommitted: Boolean = {
+    logger.trace("isCommitted called")
+    super.isCommitted || committed
+  }
 
-  override def getResponse: ServletResponse = throw new UnsupportedOperationException("getResponse is not supported")
+  override def getResponse: ServletResponse = {
```

Comment:
What are these methods? I'm assuming they are part of the contract, but what are they supposed to do?

Reply:
These methods are inherited from the `HttpServletResponseWrapper` class. They allow the user of the wrapper to get and set the underlying response. Obviously, that does not work out very well when we are keeping state in the wrapper. I imagine that is the reason we disabled the methods here.